### PR TITLE
Implement Folder and Post models

### PR DESCRIPTION
**Pull request by mccodav1**
Closes US68T90. Implements Folder and Post models with helper methods to allow easily accessing a User's Posts and Folders. Sets up serializers for the new models and the reverse relationships. Also adds some testing.

This also adds some flexibility in User creation. Currently, the Poste application passes 'name' which is split into 'first_name' and 'last_name' by Django. This has some shortcomings, however, and ideally we would move to passing 'first_name' and 'last_name' specifically. If we make these required, however, the current application will stop allowing login.

As a stopgap, the following logic applies:

```
If the application passes just 'name' (like Poste currently does) and does not provide both 'first_name' and 'last_name' it will be split into 'first_name' and 'last_name' by Django. 

If a user passes 'first_name' and 'last_name' they will be used instead, and 'name', if passed, will be ignored.

Either 'name' or 'first_name' and 'last_name' must be passed, or a ValidationError will be raised. This means that there is flexibility in what is passed.
```

### CODE REVIEW

File: PosteAPI/models.py [hunk 1]
```diff
@@ -17,7 +46,7 @@ def __str__(self):
 
 
 class Folder(models.Model):
-    title = models.CharField(max_length=100)
+    title = models.CharField(max_length=100, blank=False)
```
Comment (kmgehrma) [inline PosteAPI/models.py:49]: I agree with `title` being required.

File: PosteAPI/models.py [hunk 2]
```diff
@@ -28,9 +57,9 @@ def __str__(self):
 
 
 class Post(models.Model):
-    title = models.CharField(max_length=100, blank=True)
+    title = models.CharField(max_length=100, blank=False)
     description = models.TextField(blank=True)
-    url = models.CharField(max_length=1000, blank=False)
+    url = models.CharField(max_length=1000)
```
Comment (kmgehrma) [inline PosteAPI/models.py:62]: I think `url` should also be required:
```suggestion
    url = models.CharField(max_length=1000, blank=False)
```

**Review by techbykim — CHANGES_REQUESTED** [review]
Your validation logic for name vs. first name/last name makes a lot of sense. I would ask that the test suite be matured given the updates to the User model. Specifically that errors are thrown as expected and demonstrating a case with no space such as "Kim" - first name should be set, but last name would be an empty string or NULL.

File: PosteAPI/models.py (hunk 2 above)
Comment (mccodav1) [reply PosteAPI/models.py]: As requested.

**Comment by mccodav1** [comment]
> Your validation logic for name vs. first name/last name makes a lot of sense. I would ask that the test suite be matured given the updates to the User model. Specifically that errors are thrown as expected and demonstrating a case with no space such as "Kim" - first name should be set, but last name would be an empty string or NULL.

Tests added.

**Review by techbykim — APPROVED** [review]
Thanks for updating the tests!

**Pull request merged**